src/Layouts/Partials/Layer.vala

```diff
@@ -551,6 +578,15 @@ public class Akira.Layouts.Partials.Layer : Gtk.ListBoxRow {
         editing = false;
 
         activate ();
+
+        var new_label = entry.get_text ();
+
+        if (label.label == new_label) {
+            return;
+        }
+
+        label.label = new_label;
+        model.name = new_label;
```

Comment:
I think here you can do `label.label = mode.name = new_label;`

Reply:
Done! Thanks for the suggestion, I was not aware of this Vala feature (guess it's more of a C family thing, btw)